File: packages/ios/Sources/Components/DirectionStep/DirectionStep.figma.swift
import SwiftUI
import Figma

struct KozmosDirectionStepConnect: FigmaConnect {
    let component = KozmosDirectionStep.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1340-6763"

    @FigmaString("Instruction Text")
    var instruction: String = "Continue past the escalators"

    @FigmaString("Distance Text")
    var distance: String = "40 m"

    @FigmaString("Duration Text")
    var duration: String = "1 min"

    @FigmaEnum(
        "Type",
        mapping: [
            "Straight": DirectionType.straight,
            "Left": DirectionType.left,
            "Right": DirectionType.right,
            "Destination": DirectionType.destination
        ]
    )
    var type: DirectionType = .straight

    var body: some View {
        KozmosDirectionStep(
            type: self.type,
            instruction: self.instruction,
            distance: self.distance,
            duration: self.duration
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/DirectionStep/DirectionStep.figma.kt
package com.kozmos.components.directionstep

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1340-6763")
class KozmosDirectionStepConnect {
    @FigmaProperty(FigmaType.Text, "Instruction Text")
    val instruction: String = "Continue past the escalators"

    @FigmaProperty(FigmaType.Text, "Distance Text")
    val distance: String = "40 m"

    @FigmaProperty(FigmaType.Text, "Duration Text")
    val duration: String = "1 min"

    @FigmaProperty(FigmaType.Enum, "Type")
    val type: DirectionType = Figma.mapping(
        "Straight" to DirectionType.Straight,
        "Left" to DirectionType.Left,
        "Right" to DirectionType.Right,
        "Destination" to DirectionType.Destination,
        "LiftUp" to DirectionType.LiftUp,
        "LiftDown" to DirectionType.LiftDown,
        "EscalatorUp" to DirectionType.EscalatorUp,
        "EscalatorDown" to DirectionType.EscalatorDown,
        "StairsUp" to DirectionType.StairsUp,
        "StairsDown" to DirectionType.StairsDown,
        "LevelUp" to DirectionType.LevelUp,
        "LevelDown" to DirectionType.LevelDown,
        "Transition" to DirectionType.Transition,
        "TurnBack" to DirectionType.TurnBack
    )

    @Composable
    fun ComponentExample() {
        KozmosDirectionStep(
            type = type,
            instruction = instruction,
            distance = distance,
            duration = duration
        )
    }
}
